website/app.vue

```diff
@@ -38,53 +33,73 @@ const localize = lang => {
 }
 localize(lang)
 
-export default {
+export default defineComponent({
   name: 'App',
 
-  computed: {
-    lang() {
-      return this.$route.path.split('/')[1] || Language.CN
-    },
-    isComponent() {
-      return /^component-/.test(this.$route.name || '')
-    },
-  },
+  setup(){
+    const route = useRoute()
 
-  watch: {
-    lang(val) {
-      if (val === Language.CN) {
-        this.suggestJump()
-      }
-      localize(val)
-    },
-  },
+    const lang = computed(() => route.path.split('/')[1] || Language.CN)
 
-  mounted() {
-    localize(this.lang)
-    if (this.lang === Language.CN) {
-      this.suggestJump()
-    }
-  },
+    const isComponent = computed(() => /^component-/.test(route.name || ''))
 
-  methods: {
-    suggestJump() {
+    const _this = getCurrentInstance()
+
+    const suggestJump = () => {
       if (process.env.NODE_ENV !== 'production') return
 
       const href = location.href
       const preferGithub = localStorage.getItem('PREFER_GITHUB')
       const cnHref = href.indexOf('element-plus.gitee.io') > -1
       if (cnHref || preferGithub) return
       setTimeout(() => {
-        if (this.lang !== Language.CN) return
-        this.$confirm('建议大陆用户访问部署在国内的站点，是否跳转？', '提示')
+        if (lang.value !== Language.CN) return
+        _this.appContext.config.globalProperties.$confirm('建议大陆用户访问部署在国内的站点，是否跳转？', '提示')
```

Comment:
Maybe use `import { ElNotification } from `element-plus` could reduce this complexity.